pkg/github/github.go

```diff
@@ -348,6 +382,76 @@ func (g *GitHub) Releases(owner, repo string, includePrereleases bool) ([]*githu
 	return releases, nil
 }
 
+// GetReleaseTags returns a list of GitHub release tags for the provided
+// `owner` and `repo`. If `includePrereleases` is `true`, then the resulting
+// slice will also contain pre/drafted releases.
+func (g *GitHub) GetReleaseTags(owner, repo string, includePrereleases bool) ([]string, error) {
+	releases, err := g.Releases(owner, repo, includePrereleases)
+	if err != nil {
+		return nil, err
+	}
+
+	releaseTags := []string{}
+	for _, release := range releases {
+		releaseTags = append(releaseTags, *release.TagName)
+	}
+
+	return releaseTags, nil
+}
+
+func (g *GitHub) DownloadReleaseAssets(owner, repo string, releaseTags []string, outputDir string) error {
+	var releases []*github.RepositoryRelease
+
+	if len(releaseTags) > 0 {
+		for _, tag := range releaseTags {
+			release, _, err := g.client.GetReleaseByTag(context.Background(), owner, repo, tag)
+			if err != nil {
+				return err
+			}
+
+			releases = append(releases, release)
+		}
+	} else {
+		return errors.New("no release tags were populated")
+	}
+
+	for _, release := range releases {
+		releaseTag := release.GetTagName()
+		logrus.Infof("Download assets for %s/%s@%s", owner, repo, releaseTag)
+
+		releaseDir := filepath.Join(outputDir, owner, repo, releaseTag)
+		if err := os.MkdirAll(releaseDir, os.FileMode(0777)); err != nil {
+			return err
+		}
+
+		logrus.Infof("Writing assets to %s", releaseDir)
+
+		assets := release.Assets
+		for _, asset := range assets {
+			logrus.Infof("GitHub asset ID: %v, download URL: %s", *asset.ID, *asset.BrowserDownloadURL)
```

Comment:
Is it somehow possible that those dereferenced fields can be `nil`? I'm wondering if we can crash here.

Reply:
Good point! I'm not sure that this should ever fail, but I've added this check for good measure, which uses one of the asset methods:

```go
			if asset.GetID() == 0 {
				return errors.New("asset ID should never be zero")
			}
```

`GetID()` returns zero when the field is pointer is nil, which I think is a sufficient check to reuse here.

Reply:
Yes, I think we can stick to the getters. :+1: